plugins/basic/nexus-repository-maven/src/main/java/org/sonatype/nexus/repository/maven/internal/MavenFacetImpl.java

```diff
@@ -375,9 +376,15 @@ private void putAssetPayload(final StorageTx tx,
       if (lastModified != null) {
         formatAttributes.set(P_CONTENT_LAST_MODIFIED, lastModified.toDate());
       }
-      formatAttributes.set(P_CONTENT_ETAG, contentAttributes.get(Content.CONTENT_ETAG, String.class));
+      final String etag = contentAttributes.get(Content.CONTENT_ETAG, String.class);
+      if (!Strings.isNullOrEmpty(etag)) {
+        formatAttributes.set(P_CONTENT_ETAG, etag);
+      }
+      else {
+        formatAttributes.remove(P_CONTENT_ETAG);
+      }
     }
-    else {
+    if (formatAttributes.get(P_CONTENT_LAST_MODIFIED) == null) {
```

Comment:
Never mind - I see that the other check depends on nullity of the overall contentAttributes